File: backend/app/models/notice.py
"""
크롤링된 공지 응답 모델 (API)
"""
from datetime import datetime
from pydantic import BaseModel


class NoticeResponse(BaseModel):
    """공지 한 건 (전체 공지 탭용)"""

    id: str
    airline: str
    source_url: str
    title: str
    content_type: str
    event_start: datetime | None = None
    event_end: datetime | None = None
    is_special_deal: bool = False
    created_at: datetime

    model_config = {"from_attributes": True}

File: android/app/src/main/java/com/aerofinder/data/Notice.kt
package com.aerofinder.data

import com.google.gson.annotations.SerializedName

/**
 * 백엔드 GET /api/notices 응답 한 건 (크롤링된 전체 공지)
 */
data class Notice(
    val id: String,
    val airline: String,
    @SerializedName("source_url") val sourceUrl: String,
    val title: String,
    @SerializedName("content_type") val contentType: String,
    @SerializedName("event_start") val eventStart: String? = null,
    @SerializedName("event_end") val eventEnd: String? = null,
    @SerializedName("is_special_deal") val isSpecialDeal: Boolean = false,
    @SerializedName("created_at") val createdAt: String? = null,
)
